testGeogrid_ISCE.py

```diff
@@ -135,12 +139,57 @@ def loadMetadata(indir):
     info.prf = 1.0 / frames[0].bursts[0].azimuthTimeInterval
     info.rangePixelSize = frames[0].bursts[0].rangePixelSize
     info.lookSide = -1
+        
+    info.startingRange -= buffer * info.rangePixelSize
+    info.farRange += buffer * info.rangePixelSize
+    
     info.numberOfLines = int( np.round( (info.sensingStop - info.sensingStart).total_seconds() * info.prf)) + 1
-    info.numberOfSamples = int( np.round( (info.farRange - info.startingRange)/info.rangePixelSize)) + 1
+    info.numberOfSamples = int( np.round( (info.farRange - info.startingRange)/info.rangePixelSize)) + 1  + 2 * buffer
     info.orbit = getMergedOrbit(frames)
 
     return info
 
+def loadParsedata(indir,buffer=0):
+    '''
+    Input file.
+    '''
+    import os
+    import numpy as np
+    import isce
+    from isceobj.Sensor.TOPS.Sentinel1 import Sentinel1
+    
+
+    frames = []
+    for swath in range(1,4):
+        rdr=Sentinel1()
+        rdr.configure()
+#        rdr.safe=['./S1A_IW_SLC__1SDH_20180401T100057_20180401T100124_021272_024972_8CAF.zip']
+        rdr.safe=[indir]
+        rdr.output='reference'
+        rdr.orbitDir='/Users/yanglei/orbit/S1A/precise'
+        rdr.auxDir='/Users/yanglei/orbit/S1A/aux'
+        rdr.swathNumber=swath
+        rdr.polarization='hh'
```

Comment:
This is also hard-coded and should be parsed from the SLC name. Can use a function like:
```python
from typing import Tuple
from pathlib import Path


def get_polarizations(s1_safe: str) -> Tuple[str]:
    mapping = {
        'SH': ('hh',),
        'SV': ('vv',),
        'DH': ('hh', 'hv'),
        'DV': ('vv', 'vh'),
    }
    key = Path(s1_safe).name[14:16]
    return mapping[key]
```

and selected like:
```suggestion
        rdr.polarization=get_polarizations(indir)[0]
```